src/v/raft/tests/raft_fixture.h

```diff
@@ -374,59 +374,67 @@ class raft_fixture
 
     template<typename Func>
     auto retry_with_leader(
-      int retry_count, model::timeout_clock::time_point deadline, Func&& f) {
+      model::timeout_clock::time_point deadline,
+      std::chrono::milliseconds backoff,
+      Func&& f) {
         using futurator
           = ss::futurize<std::invoke_result_t<Func, raft_node_instance&>>;
         using ret_t = futurator::value_type;
         struct retry_state {
             ret_t result = errc::timeout;
-            int retries_left;
+            int retry = 0;
         };
         return ss::do_with(
-          retry_state{.retries_left = retry_count},
-          [this, deadline, f = std::forward<Func>(f), retry_count](
+          retry_state{},
+          [this, deadline, f = std::forward<Func>(f), backoff](
             retry_state& state) mutable {
               return ss::do_until(
-                       [&state] { return state.retries_left <= 0; },
+                       [&state, deadline] {
+                           return model::timeout_clock::now() > deadline
+                                  || state.result.has_value();
+                       },
                        [this,
                         &state,
                         f = std::forward<Func>(f),
                         deadline,
-                        retry_count]() mutable {
+                        backoff]() mutable {
                            vlog(
                              _logger.info,
-                             "Executing action with leader, retries left {}/{}",
-                             state.retries_left,
-                             retry_count);
+                             "Executing action with leader, current retry: {}",
+                             state.retry);
                            if (model::timeout_clock::now() > deadline) {
```

Comment:
This statement isn't needed now that it's part of the `do_until` condition right?

Reply:
yes, removing

Reply:
done